.github/workflows/scripts/comments.mjs

```diff
@@ -1,14 +1,67 @@
+import { readFile } from 'node:fs/promises'
+
+/**
+ * Posts the content of a diff as a comment on a Github issue
+ *
+ * @param {GithubActionContext} githubActionContext
+ * @param {number} issueNumber
+ * @param {DiffComment} diffComment
+ */
+export async function commentDiff (
+  githubActionContext,
+  issueNumber,
+  { path, title, markerText }
+) {
+  // Read diff from previous step
+  const diffText = await readFile(path, 'utf8')
+
+  // Add or update comment on PR
+  try {
+    await comment(
+      githubActionContext,
+      issueNumber,
+      markerText,
+      renderDiffBodyText(title, diffText)
+    )
+  } catch {
+    await comment(
+      githubActionContext,
+      issueNumber,
+      markerText,
+      `## ${title}\n` +
+        `The diff could not be posted as a comment. You can download it from the [workflow artifacts](${githubActionRunUrl(
+          githubActionContext.context.runId
+        )}#artifacts).`
+    )
+  }
+}
+
 /**
- * @param {object} githubContext - Github Action context
- * @param {import('@octokit/rest').Octokit} githubContext.github
- * @param {import('@actions/github').context} githubContext.context
- * @param {number} issueNumber - GitHub issue or PR number
+ * @param {string} title - The title for the comment
+ * @param {string} diffText - The diff content
+ * @returns {string} The content for the comment body
+ */
+function renderDiffBodyText (title, diffText) {
```

Comment:
I've added a special case when the diff is empty so the comment does not look broken, making us wonder if something went wrong.